src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs

```diff
@@ -48,13 +48,17 @@ public override void ExecuteCmdlet()
         {
             try
             {
-                if (!string.IsNullOrEmpty(this.BackupStorageRedundancy))
+                if (this.BackupStorageRedundancy.HasValue)
                 {
                     UpdateVaultStorageTypeRequest vaultStorageRequest = new UpdateVaultStorageTypeRequest();
                     vaultStorageRequest.Properties = new StorageTypeProperties();
-                    vaultStorageRequest.Properties.StorageModelType = BackupStorageRedundancy;
+                    vaultStorageRequest.Properties.StorageModelType = BackupStorageRedundancy.ToString();
                     AzureOperationResponse storageResponse = RecoveryServicesClient.UpdateVaultStorageType(this.Vault.ResouceGroupName, this.Vault.Name, vaultStorageRequest);
                 }
+                else
+                {
+                    this.HandleException(new Exception("You need to provide the BackupStorageRedundancy input parameter"));
```

Comment:
yes, pls add in resx file. Also make sure to run all tests and passing before checkin.